### PR TITLE
Add Athletics API

### PR DESCRIPTION
**Pull request by kashav**
- List / show / search
- Filter (incl. matched events 🎉 )
  - Dates take input as `Y,m,d,H,M,S` (like a JavaScript `Date`)
- Tests

Some filter examples:

`/1.0/athletics/filter?q=title:"rock climbing" AND date:"2016,04,04"`

``` js
[
  {
    "id":"04SC",
    "date":"2016-04-04T00:00:00.000Z",
    "campus":"UTSC",
    "events":[...],
    "matched_events":[{
      "title":"Rock Climbing Club",
      "location":"Climbing Wall",
      "building_id":"208",
      "start_time":"2016-04-04T18:00:00.000Z",
      "end_time":"2016-04-04T20:30:00.000Z"
    }]
  }
]
```

`/1.0/athletics/filter?q=start:<"2016,04,01,12"`

``` js
[
  {
    "id":"01M",
    "date":"2016-04-01T00:00:00.000Z",
    "campus":"UTM",
    "events":[...],
    "matched_events":[{
      "title":"Length Swim",
      "location":"Pool",
      "building_id":"332",
      "start_time":"2016-04-01T11:00:00.000Z",
      "end_time":"2016-04-01T13:00:00.000Z"
    }]
  }
]
```

https://github.com/cobalt-uoft/cobalt/issues/57

### CODE REVIEW

File: src/api/athletics/routes/filter.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    let d = dateValues.concat(new Array(7 - dateValues.length).fill(0))
+
+    // Months[1] start at index 0 (Jan->0, Dec->11), hours[3] are altered for EST
+    let date = new Date(d[0], d[1]-1, d[2], d[3]-4, d[4], d[5], d[6], d[7])
+
+    if (isNaN(date)) {
+      response.isValid = false
+      response.error = new Error('Invalid date parameter.')
+      response.error.status = 400
+      return response
```
Comment (kashav) [inline src/api/athletics/routes/filter.js:167]: For invalid dates, I ignore any other queries and throw an error. Should we just ignore the `date` query in that case and continue with the others or is this fine?

Think of a scenario like:

`/1.0/athletics?filter?q=date:"invalid date" AND title:"basketball"`

Response would be something like

``` json
{
  "error": {
    "code": 400,
    "message": "Invalid date parameter."
  }
}
```

Comment (qasim) [inline src/api/athletics/routes/filter.js:167]: It should fail right away and serve the `400`. User should be responsible for ensuring their query is 100% valid.

**Comment by kashav** [comment]
Not too sure why that test was failing, passed when I tested locally. Removed for the time being.

**Comment by qasim** [comment]
I'll start testing this first thing tomorrow. From my first glance it's looking good!

File: src/db/index.js [hunk 2]
```diff
@@ -56,14 +69,15 @@ db.sync = () => {
...
+  db.update('athletics')
 }
 
 db.check = (callback) => {
   let options  =  {
     host: 'api.github.com',
     port: 443,
     path: '/repos/cobalt-uoft/datasets/git/refs/heads/master',
-    headers: {'user-agent': 'cobalt-uoft/0.4.2'}
+    headers: {'user-agent': `cobalt-uoft/${version}`}
```
Comment (qasim) [inline src/db/index.js:80]: this is beautiful

**Comment by qasim** [comment]
I'm guessing there's no nice way to have the date outputted localized for Eastern timezone? I see Mongo converts it to the zero-offset UTC.

**Comment by kashav** [comment]
I think something like [this](https://docs.mongodb.org/manual/tutorial/model-time-data/) could work.

We'd just subtract the offset from the datetime when converting date strings (after importing the dataset).

**Comment by qasim** [comment]
Let's leave it as is for now and can revisit later. I've a feeling we'll run into more datetime problems as the project progresses. GG time.

**Pull request merged**